### DISCUSSION TITLE
JS vs Python for scripting

### DISCUSSION THREAD

**Discussion by emarj** [chapar-rest/chapar#33] [Ideas]
First of all congratulation for this awesome project, I am really excited to use it.

I am curious on the choice of using Python as scripting language instead of Javascript. What went into that decision? Could you elaborate a bit on this?

I am not a fan of JS, and I would rather use Python instead, in general. Nonetheless JS is the de-facto standard of the web, so it could make more sense to use it in such a tool. Moreover (I think), it should be easier to integrate "inside" Go, but I could be wrong on this. What do you think?

Congrats and keep up with the good work

**Comment by mirzakhany** [comment]
Hi @emarj and thanks for raising this question.
To be honest I have chosen Python as part of the plan because I know it better. And Im terrible in JS 😅
That being said I’m up for making both available and let user choose. 
My plan is to make some sort of server in Chapar that send some meta data to an external process and take answer back. In my book that process can be anything.

happy to discuss it even further 🙏

**Reply by emarj** [reply]
Ok, with this "server" approach the choice of the "client" language would be of secondary importance I guess.

I thought you meant to have snippets of code interpreted by Chapar. Like for example using [goja](https://github.com/dop251/goja) or [otto](https://github.com/robertkrimen/otto) (for JS).
This could be easier to implement, but maybe more limited.

Of course it depends on the functionalities you want to offer.

But again, this was just a curiosity on my part 😅

**Reply by mirzakhany** [reply]
I've seen those projects, and the reason I want to avoid them is their limitations. Imagine if a user needs a specific library to perform post/pre-request actions; an internal interpreter would be limiting. On the other hand, having an external process, just like a normal Python process, gives users the choice of installing their required dependencies.

My plan is to have a virtual environment set up with Chapar, where users can add packages to it. This idea is not fully formed in my head, but I kinda know where to start 😄. Please don't hesitate to share your ideas, thoughts, and feedback 🙏
